src/sentry/snuba/discover.py

```diff
@@ -69,6 +69,9 @@
 
 class EventsMeta(TypedDict):
     fields: Dict[str, str]
+    datasetReason: Optional[str]
+    isMetricsData: Optional[bool]
+    isMetricsExtractedData: Optional[bool]
```

Comment:
are these `Optional[str]` (aka, either a `str` or `None`) or are these `NotRequired[str]` (either a `str` or it is missing) ?

Reply:
@asottile-sentry not required. Thanks for catching it! I was not aware of it